Database/MySQL/Base/Types.hsc

```diff
@@ -132,11 +169,74 @@ toType v = IntMap.findWithDefault oops (fromIntegral v) typeMap
       , ((#const MYSQL_TYPE_VAR_STRING), VarString)
       , ((#const MYSQL_TYPE_STRING), String)
       , ((#const MYSQL_TYPE_GEOMETRY), Geometry)
-#if defined(MYSQL_TYPE_JSON)
```

Comment:
This seems to actually be causing some issues and is why i ended up on this journey in the first place.

I think it is safe to remove since MYSQL_TYPE_JSON has been supported since 2015